aten/src/ATen/native/mps/operations/ReduceOps.mm

```diff
@@ -931,6 +935,88 @@ static void argmax_argmin_out_mps(const Tensor& input_t,
   }
 }
 
+static void cum_max_min_helper_mps(const Tensor& input,
+                                   Tensor& values,
+                                   Tensor& indices,
+                                   int64_t dim,
+                                   MPSReductionType reduction_type) {
+  if (!is_macos_13_or_newer()) {
+    TORCH_WARN_ONCE("MPS: cummax_helper_mps op is supported natively starting from macOS 13.0. ",
+                    "Falling back on CPU. This may have performance implications.");
+
+    Tensor values_cpu = at::empty({input.sizes()}, input.options());
+    Tensor indices_cpu = at::empty({input.sizes()}, input.options().dtype(kLong));
```

Comment:
Nit (why not use `empty_like`? Also, shouldn't values and indices already be of an expected size? And if they are not, we should resize
```suggestion
    auto values_cpu = at::empty_like(input, TensorOptions(kCPU)));
    auto indices_cpu = at::empty(input, TensorOptions(kCPU).dtype(kLong));
    auto input_cpu = input.to("cpu");
```